common-test/src/main/java/io/strimzi/test/StrimziRunner.java

```diff
@@ -294,25 +295,52 @@ protected void after() {
         return last;
     }
 
+    private String changeOrgAndTag(String image, String newOrg, String newTag) {
+        return image.replaceFirst("^strimzi/", newOrg + "/").replaceFirst(":[^:]+$", ":" + newTag);
+    }
+
     private Statement withClusterController(Annotatable element,
                                     Statement statement) {
         Statement last = statement;
         for (ClusterController cc : annotations(element, ClusterController.class)) {
+            List<String> yamls = Arrays.stream(new File(CC_INSTALL_DIR).listFiles()).sorted().map(f -> getContent(f, node -> {
+                // Change the docker org of the images in the 04-deployment.yaml
+                if ("04-deployment.yaml".equals(f.getName())) {
+                    String dockerOrg = System.getenv().getOrDefault("DOCKER_ORG", "strimzici");
+                    String dockerTag = System.getenv().get("DOCKER_TAG");
+                    if (dockerTag == null || dockerTag.isEmpty()) {
+                        throw new RuntimeException("DOCKER_TAG environment variable not set");
+                    }
+                    JsonNode containerNode = node.get("spec").get("template").get("spec").get("containers").get(0);
```

Comment:
Is it a real case that we can have several containers and we need to apply a new value of docker org and tag for all of them?

Reply:
@serrss I'm not completely sure I understand what you're asking. This bit of code is basically modifying the `examples/install/cluster-controller/04-deployment.yaml`. That specifies a Deployment, which has a `template` for the pods the deployment will manage. Each pod can have 1 or more containers, but we only use one -- the cluster controller. So the first bit of code changes the `image` for that container. Then the CC will deploy Kafka and Zookeeper StatefulSets, etc, and so we need to configure the CC to use the images we want: We do that by passing it env vars, 1 for each of the things it deploys.

Reply:
@tombentley Sorry for unclear question =( The question was related to `.get(0)` and what will be if we have 2 elements in the array of JSON key `containers`.

Reply:
@serrss here we are referring to the cluster controller Deployment which (we know) is made by just one container.